lighthouse-core/audits/preload-lcp.js

```diff
@@ -0,0 +1,185 @@
+/**
+ * @license Copyright 2020 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const URL = require('../lib/url-shim.js');
+const Audit = require('./audit.js');
+const i18n = require('../lib/i18n/i18n.js');
+const MainResource = require('../computed/main-resource.js');
+const LanternLCP = require('../computed/metrics/lantern-largest-contentful-paint.js');
+const LoadSimulator = require('../computed/load-simulator.js');
+const UnusedBytes = require('./byte-efficiency/byte-efficiency-audit.js');
+
+const UIStrings = {
+  /** Title of a lighthouse audit that tells a user to preload an image in order to improve their LCP time. */
+  title: 'Preload Largest Contentful Paint image',
+  /** Description of a lighthouse audit that tells a user to preload an image in order to improve their LCP time.  */
+  description: 'Consider preloading the image used by ' +
+    'the LCP element in order to improve your LCP time.',
+};
+
+const str_ = i18n.createMessageInstanceIdFn(__filename, UIStrings);
+
+class PreloadLCPAudit extends Audit {
+  /**
+   * @return {LH.Audit.Meta}
+   */
+  static get meta() {
+    return {
+      id: 'preload-lcp',
+      title: str_(UIStrings.title),
+      description: str_(UIStrings.description),
+      requiredArtifacts: ['traces', 'devtoolsLogs', 'URL', 'TraceElements'],
+      scoreDisplayMode: Audit.SCORING_MODES.NUMERIC,
+    };
+  }
+
+  /**
+   *
+   * @param {LH.Artifacts.NetworkRequest} request
+   * @param {LH.Artifacts.NetworkRequest} mainResource
+   * @param {Array<LH.Gatherer.Simulation.GraphNode>} initiatorPath
+   * @param {string|undefined} lcpImageSource
+   * @return {boolean}
+   */
+  static shouldPreloadRequest(request, mainResource, initiatorPath, lcpImageSource) {
+    const mainResourceDepth = mainResource.redirects ? mainResource.redirects.length : 0;
+
+    // If it's not the request for the LCP image, don't recommend it.
+    if (request.url !== lcpImageSource) return false;
+    // If it's already preloaded, no need to recommend it.
+    if (request.isLinkPreload) return false;
+    // It's not a request loaded over the network, don't recommend it.
+    if (URL.NON_NETWORK_PROTOCOLS.includes(request.protocol)) return false;
+    // It's not at the right depth, don't recommend it.
+    if (initiatorPath.length < mainResourceDepth + 1) return false;
```

Comment:
```suggestion
    // It's already discoverable from the main document, don't recommend it.
    if (initiatorPath.length <= mainResourceDepth) return false;
```

feels a little clearer :) I'm not sure what if this is doing what we want it to right now though. If the main resource didn't redirect, then `mainResourceDepth` will be 0 and `initiatorPath.length` is guaranteed to be `>=1` for everything but the root document itself, so this doesn't seem like it would prevent anything?

I think we should eventually be able to relax this bit once the graph is brought into play and we can suggest preloading even if the image isn't at a deeper discovery level if the request was just low priority and issued late.